src/App.js

```diff
@@ -1,26 +1,104 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react';
+// import logo from './logo.svg';
 import './App.css';
+import FoodBox from './components/FoodBox';
+import foods from './foods.json';
+import AddFoodItem from './components/AddFoodItem';
+import 'bootstrap/dist/css/bootstrap.css';
+import { Button } from 'react-bootstrap';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+  state = {
+    foodItems: foods,
+    showForm: false,
+    filteredFood: foods,
+  };
+
+  //add a food item
+  handleAdd = (event) => {
+    event.preventDefault();
+
+    const { foodItems, filteredFood } = this.state;
+
+    //to obtain the value passed by the user in the form
+    let foodName = event.target.name.value;
+    let foodCalo = event.target.calories.value;
+    let foodImage = event.target.image.value;
+
+    //console.log("event.target.name.value is: ", event.target.name.value)
+    let newFoodAdded = {
+      name: foodName,
+      calories: foodCalo,
+      image: foodImage,
+    };
+
+    this.setState({
+      foodItems: [...foodItems, newFoodAdded],
+      filteredFood: [...filteredFood, newFoodAdded],
+      showForm: false,
+    });
+  };
+
+  //make the form disappear when the user click submit
+  handleFormShow = () => {
+    this.setState({
+      showForm: true,
+    });
+  };
+
+  //to search for a food item
+  handleSearch = (event) => {
+    console.log('event.target.value is: ', event.target.value);
+
+    let searchItem = event.target.value;
+    const { foodItems } = this.state;
+
+    let filteredFood = foodItems.filter((foodItem) => {
+      return foodItem.name.includes(searchItem);
+    });
+
+    this.setState({
+      filteredFood: filteredFood,
+    });
+  };
+
+
+  handleAddFoodToList = () => {
+    
+  }
+
+
+
+
+  render() {
+    const { foodItems, showForm, filteredFood } = this.state;
+
+    return (
+      <div>
+        {/* to show or hide the adding food form */}
+        {showForm ? (
+          <AddFoodItem onAdd={this.handleAdd} />
+        ) : (
+          <Button variant="outline-info" onClick={this.handleFormShow}>
+            Show
+          </Button>
+        )}
+        {/* search bar */}
+        <input
+          onChange={this.handleSearch}
+          type="text"
+          placeholder="Search"
+        ></input>
+
+        <div class="columns">
+          <div class="column is-two-thirds"><FoodBox food={filteredFood} /></div>
+          <div class="column"><h2>Today's foods</h2></div>
```

Comment:
Try to put nested elements like the ones inside divs, components or h tags in different lines.